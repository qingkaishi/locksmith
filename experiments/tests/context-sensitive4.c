#include <stdio.h>		/* Input/Output */
#include <pthread.h>		/* POSIX Threads */
  pthread_mutex_t L1, L2;
int x, y;

void
munge (pthread_mutex_t * l, int *p)
{
  pthread_mutex_lock (l);
  *p = 3;
  pthread_mutex_unlock (l);
}

void*
f (void *ptr)
{
  munge (&L1, &x);
  munge (&L2, &y);
  return NULL;
}

int
main ()
{
  pthread_t thread1, thread2;

  pthread_mutex_init (&L1, NULL);
  pthread_mutex_init (&L2, NULL);

  pthread_create (&thread1, NULL, f, NULL);
  pthread_create (&thread2, NULL, f, NULL);
}
