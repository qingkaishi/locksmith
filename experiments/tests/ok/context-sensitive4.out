************************* STARTING *************************

shared: &x:context-sensitive4.c:4
        &y:context-sensitive4.c:4
&x:context-sensitive4.c:4 is protected by:
  *munge.l:context-sensitive4.c:17
  L1:context-sensitive4.c:3
  concrete L1:context-sensitive4.c:27
&y:context-sensitive4.c:4 is protected by:
  *munge.l:context-sensitive4.c:18
  L2:context-sensitive4.c:3
  concrete L2:context-sensitive4.c:28
*************************** DONE ***************************
